pennylane/ftqc/pauli_tracker.py

```diff
@@ -235,3 +245,242 @@ def commute_clifford_op(clifford_op: Operator, xz: List[Tuple[int, int]]) -> Lis
+    for op in ops:
+        for wire in op.wires:
+            _wires_used[wire] += 1
+
+        gate_offset = 4 if op.num_wires == 1 else 13
+        ms = mid_meas[mid_meas_offset : mid_meas_offset + gate_offset]
+        by_op = []
+        if isinstance(op, (S, H)):
+            by_op = [_single_xz_corrections(op, *ms)]
+        elif isinstance(op, CNOT):
+            by_op = _cnot_xz_corrections(ms)
+        elif isinstance(op, (RZ, RotXZX)):
+            if _wires_used[op.wires[0]] > 1:
+                raise NotImplementedError(
+                    "Current implementation only support that one non-Clifford gate comes before Clifford and Pauli gates for each wire."
+                )
+            by_op = [_single_xz_corrections(op, *ms)]
+        elif isinstance(op, _PAULIS):
+            continue
+        else:
+            raise NotImplementedError(f"{op.name} is not supported.")
+
+        by_ops.append(by_op)
+
+        mid_meas_offset += gate_offset
+
+    # To use list in a stack manner
+    by_ops.reverse()
+    return by_ops
+
+
+def _get_xz_record(tape: QuantumScript, by_ops: List[Tuple[int, int]], ops: List[Operator]):
+    """Commutate/merge the Pauli/byproduct ops of a Clifford circuit.
+
+    Args:
+        tape (QuantumScript): A quantum tape.
+        by_ops (list): List of byproduct operators for Clifford gates
+        ops (list): List of Clifford/Pauli/StatePrep operations.
+
+    Return:
+        The final recorded x and z for each wire.
+    """
+
+    num_wires = max(tape.wires) + 1
+
+    x_record = math.zeros(num_wires, dtype=np.uint8)
+    z_record = math.zeros(num_wires, dtype=np.uint8)
+
+    for op in ops:
+        wires = list(op.wires)
+
+        # Get the recorded xz
+        xz = [(x_record[wire], z_record[wire]) for wire in wires]
+
+        # Updated xz
+        new_xz = []
+        if isinstance(op, _NON_CLIFFORD_GATES_SUPPORTED):
+            # Branch for non-Clifford gates
+            new_xz.append(by_ops.pop()[0])
+        elif isinstance(op, _CLIFFORD_GATES_SUPPORTED):
+            # Branch for Clifford gates
+            # Step 1: Commutate the recorded xz with the Clifford gate to a new xz.
+            xz_commutated = commute_clifford_op(op, xz)
+
+            # Step 2: Merge the new xz with the byproduct by_op
+            by_op = by_ops.pop()
+            for _by_op, _xz_comm in zip(by_op, xz_commutated):
+                new_xz.append(math.bitwise_xor(_by_op, _xz_comm))
+        else:  # branch for Paulis
+            # Commutate step is skipped.
+            # Get the new xz by merging the recorded xz with the Pauli ops directly.
+            new_xz.append(math.bitwise_xor(pauli_to_xz(op), xz[0]))
+        # Assign the updated the xz to the x, z record
+        for idx, wire in enumerate(wires):
+            x_record[wire], z_record[wire] = new_xz[idx]
+
+    return x_record, z_record
+
+
+def _get_sample_corrections(tape: QuantumScript, x_record: math.array, measurement_vals: List):
```

Comment:
Minor, but the name here sounds like it returns the corrections (`x_at_target`) rather than the corrected measurements:

```suggestion
def _correct_samples(tape: QuantumScript, x_record: math.array, measurement_vals: List):
```